docs/proposals/kubelet-oci-runtime.md

```diff
@@ -0,0 +1,87 @@
+`oci-runtime`.
+
+### Runtime API
+The implementation will make use of the kubelet runtime API (https://github.com/kubernetes/kubernetes/pull/25899). The
+kubelet will implement the client API and the container runtimes including the one for OCI will implement the server
+API. The separation of client/server allows for faster iteration on the runtime integration development.
+
+### Image management
+For the first release, oci-runtime will continue to use docker-engine for managing images.  The image management
+functionality will be separated from the runtime functionality so each could have different implementations which could
+potentially be switched.  It will be ideal to support the OCI image specification for images once it reaches v1.0. In
+addition to the docker image format, other packaging formats like tar files and docker image tar files, etc will also be
+supported. Image management requires quite a bit of flexibility in various fronts like transport, content verification,
+etc., and this proposal will not delve into those aspects. This proposal essentially opens up a door for easily
+extending support for alternate image formats.  Image managers are expected to expose the disk usage of each image and
+the overall usage of disk for all images.  To provide better security, it is possible to chown images of each pod to a
+separate uid and use user namespaces to have containers in that pod use that uid. This added security comes with a cost,
+because it will no longer be possible to share images between different pods and layering of images will no longer make
+any sense. 
+
+### Lifecycle
+While the integration will implement the kubelet runtime API, we will have the oci-runtime manage the lifecycle of
+containers. The lifecycle of the container will not be tied to that of the oci-runtime.  Since the oci-runtime can be
+restarted while the containers are running, there needs to be a reliable mechanism to collect the exit code of
+containers.  One obvious choice is to use systemd as the parent of these containers.  Another option is to run a wrapper
+process and have the wrapper collect the exit code of the container and make it available to the oci-runtime via a named
+pipe.  The oci-runtime will provide plugins for lifecycle management and both of the options mentioned above will be
+made available.  The kubelet will interact with the runtime to control the lifecycle using the runtime API. Instead of
+the declarative SyncPods API, the kubelet will have much finer control over the pod/container lifecycles. 
+
+### Logging
+Logging will work by using files to redirect stdout / stderr of the containers. Other higher level drivers could build
+on top of files. Using files allow using disk quotas, impose Disk IO limits, and also avoid bottlenecks such as a SPOF
+daemons. This approach also let’s Kubelet manage the lifecycle of logs with policies. Whenever there is disk pressure,
```

Comment:
> Is the SPOF daemon a reference to journald?

Docker daemon primarily. 

> I think we should have stronger language here about making a long term goal to support structured logging out of containers and pods.

Why does k8s have to bother itself with logging formats? Most users are going to stick to formats that can be understood by logging agents like fluentd, splunk, etc, right?